### PR TITLE
contrib: remove builder keys

### PR DESCRIPTION
**Pull request by fanquake**
This has been superseded by adding a builder-keys/ directory in
guix.sigs, where the presence of keys, and validity of signatures
is checked. Preventing issues like missing keys or invalid signatures.

New (or exisiting) Guix builders can add their key in the next PR
they open adding attestations.

Related to issues like #26566, #26563.

Also follows up with the comment here: https://github.com/bitcoin/bitcoin/pull/26565#issuecomment-1326053939.

### CODE REVIEW

**Review by maflcko — APPROVED** [review]
ACK

**Comment by fanquake** [comment]
Pushed to add svanstaa (670BC460DC8BF5EEF1C3BC74B14CC9F833238F85), which has just been uploaded to keyservers: https://github.com/bitcoin-core/guix.sigs/pull/463#issuecomment-1330658995.

**Comment by fanquake** [comment]
The key for kvaciral is here: https://github.com/Kvaciral/kvaciral. So I've included that as well.

File: contrib/builder-keys/keys.txt [hunk 1]
```diff
@@ -1,57 +1,34 @@
...
 F4FC70F07310028424EFC20A8E4256593F177720 Oliver Gugger (guggero, Oliver Gugger)
-D62A803E27E7F43486035ADBBCD04D8E9CCCAC2A Paul Rabahy (prab)
-37EC7D7B0A217CDB4B4E007E7FAB114267E4FA04 Peter Todd (petertodd)
 D762373D24904A3E42F33B08B9A408E71DAAC974 Pieter Wuille [Location: Leuven, Belgium] (sipa)
 133EAC179436F14A5CF1B794860FEB804E669320 Pieter Wuille (sipa)
+2F78ACF677029767C8736F13747A7AE2FB0FD25B satsie (satsie)
 6A8F9C266528E25AEB1D7731C2371D91CB716EA7 Sebastian Falbesoner (theStack)
-A8FC55F3B04BA3146F3492E79303B33A305224CB Sebastian Kung (TheCharlatan)
 ED9BDF7AD6A55E232E84524257FF9BDBCC301009 Sjors Provoost (sjors)
+4DAF18FE948E7A965B30F9457E296D555E7F63A7 Sjors Provoost (sjors)
```
Comment (Sjors) [inline contrib/builder-keys/keys.txt:28]: I used this key once (?) by mistake. Though I guess there's no harm in adding it.

File: contrib/builder-keys/keys.txt [hunk 2]
```diff
@@ -1,57 +1,34 @@
...
-CA03882CB1FC067B5D3ACFE4D300116E1C875A3D MeshCollider (meshcollider)
 E777299FC265DD04793070EB944D35F9AC3DB76A Michael Ford (fanquake)
-AD5764F4ADCE1B99BDFD179E12335A271D4D62EC Michael Tidwell (miketwenty1)
-9692B91BBF0E8D34DFD33B1882C5C009628ECF0C Michagogo (michagogo)
-C57E4B42223FDE851D4F69DD28DF2724F241D8EE midnightmagic (midnightmagic)
+2CBBF208E594BF439B5F276C7465CFFF6793242E Niklas Gögge (dergoegge)
 F4FC70F07310028424EFC20A8E4256593F177720 Oliver Gugger (guggero, Oliver Gugger)
-D62A803E27E7F43486035ADBBCD04D8E9CCCAC2A Paul Rabahy (prab)
-37EC7D7B0A217CDB4B4E007E7FAB114267E4FA04 Peter Todd (petertodd)
 D762373D24904A3E42F33B08B9A408E71DAAC974 Pieter Wuille [Location: Leuven, Belgium] (sipa)
```
Comment (MarcoFalke) [inline contrib/builder-keys/keys.txt:23]: This expired long before guix was a thing, so should be removed, no?

**Comment by Sjors** [comment]
The "gitian era" isn't that long ago. Maybe keep the keys of people who made recent commits or other contributions? In any case, if they're added back later, we should remember to check this commit to see if it's a key we know or a new one.

**Review by maflcko — COMMENTED** [review]
According to the readme:

> Add your key to the list if you provided Guix attestations for two major or minor releases of Bitcoin Core.

So I am wondering if we should add keys for one-off signatures here?

**Comment by fanquake** [comment]
> So I am wondering if we should add keys for one-off signatures here?

I'm not sure. If we don't have the keys, people will just complain they are missing. Historically, people don't seem to be good at adding their own keys here either. We also don't want to be running a faux keyserver.

**Comment by Sjors** [comment]
> So I am wondering if we should add keys for one-off signatures here?

I would keep them if they're already here. If we want to lower the threshold for adding them (or even add them "non interactively"), that seems like a separate issue.

File: contrib/builder-keys/keys.txt (hunk 2 above)
Comment (fanquake) [reply contrib/builder-keys/keys.txt]: Yep, dropped.

**Comment by MarcoFalke** [comment]
> I'm not sure. If we don't have the keys, people will just complain they are missing. Historically, people don't seem to be good at adding their own keys here either. We also don't want to be running a faux keyserver.

Well, if you think it is important to always have all keys, then your are running a keyserver (sort of). It will never be possible to have all keys at all times be listed, non-revoked and non-expired. So attempting that shouldn't be a goal.

However, I wonder if this whole file can be removed and be replaced by having the full keys in the attestations repo, see also https://github.com/bitcoin-core/guix.sigs/issues/133

**Comment by Sjors** [comment]
There's (probably) not as much scrutiny on the guix.sigs repo as on this one, so at minimum we should keep (some) key ids here. For someone who has no developers inside their web of trust, the list here is a nice place to start.

If someone downloads the binary along with source code (e.g. via the Torrent), then they'll have this file, but not the guix.sigs repo.

**Comment by RandyMcMillan** [comment]
Maybe refactor tooling to use the guix.sigs as a subtree/submodule - then it can be pinned to a commit with every release as needed? This would be more deterministic?

**Comment by MarcoFalke** [comment]
> There's (probably) not as much scrutiny on the guix.sigs repo as on this one

This pull request is adding key fingerprints because they happened to be used once to sign and also happened to be on the internet somewhere. I don't think this is any different from just doing https://github.com/bitcoin-core/guix.sigs/issues/133

**Comment by vertiond** [comment]
My github username is `vertiond`. Could you adjust my name in parentheses to  `vertion (vertiond)`?

**Comment by luke-jr** [comment]
Seems like we should ping @schildbach @btcdrak @theuni @droark @cisba @fivepiece @Fuzzbawls @jhfrontz @jl2012 @jtimon @kallewoof @kristapsk @niftynei @MarcoFalke @TheBlueMatt @meshcollider @miketwenty1 @Michagogo @midnightmagic @petertodd @TheCharlatan @wtogami @willyko to see if any of them have interest in doing Guix builds.

**Comment by TheCharlatan** [comment]
> Seems like we should ping ...  to see if any of them have interest in doing Guix builds.

Yes, I'll contribute some.

**Comment by jhfrontz** [comment]
> Hi! Thanks for the reminder. Been meaning to get that going. I'll do it sometime this week.

Ditto.

**Comment by kallewoof** [comment]
Interest: yes, time: limited. If my speedy participation is required, you may remove my key.

**Comment by miketwenty1** [comment]
@luke-jr thanks for the ping. 
I will be interested in participating over on bitcoin-core/guix.sigs repo early next year on future builds. Seems that this is now easy enough for a caveman to do it.. count me in.

**Comment by petertodd** [comment]
Feel free to remove mine. If I do a guix build I need to update my key anyway.

**Comment by cisba** [comment]
I'm interested in doing Guix builds.
I'll setup my env during holiday.

**Comment by MarcoFalke** [comment]
Alternative in https://github.com/bitcoin-core/guix.sigs/pull/535

**Comment by MarcoFalke** [comment]
ACK 1b1a5bea7ddefa3a386733bef9a613013de5fbc7

**Review by hebasto — APPROVED** [review]
ACK e6864fa157d75d8ae2b2f56620b019bde2355a24, modulo s/update/remove/ in the PR tittle.

**Comment by thesamesam** [comment]
It'd be great for downstreams if there was a list of keys for _committers_ and people who sign the release source code tarball.

**Pull request closed without merging**